### DISCUSSION TITLE
Home offsets (M206)

### DISCUSSION THREAD

**Discussion by lagnat** [mriscoc/Ender3V2S1#1189] [General]
The configurator has an option to enable Home Offsets but it seemingly enables workspace offsets.  Shouldn't it be:

```diff --git a/_features/HomeOffs.json b/_features/HomeOffs.json
index 49629a3..d31b3a4 100644
--- a/_features/HomeOffs.json
+++ b/_features/HomeOffs.json
@@ -4,9 +4,9 @@
 "Configuration_adv.h" : [
   {
     "op": "Disable",
-    "searchfor": "NO_WORKSPACE_OFFSETS"
+    "searchfor": "NO_HOME_OFFSETS"
   }
 ],
 "Version.h" : [
 ]   
-}
\ No newline at end of file
+}
```

**Comment by lagnat** [comment]
And if the answer is yes, then:

```
diff --git a/Marlin/src/lcd/e3v2/proui/dwin.cpp b/Marlin/src/lcd/e3v2/proui/dwin.cpp
index 8528fa5..15b751a 100644
--- a/Marlin/src/lcd/e3v2/proui/dwin.cpp
+++ b/Marlin/src/lcd/e3v2/proui/dwin.cpp
@@ -2262,7 +2262,7 @@ void applyMove() {
 #endif
 
 #if HAS_HOME_OFFSET
-  void applyHomeOffset() { set_home_offset(hmiValue.select, menuData.value / MINUNITMULT); }
+  void applyHomeOffset() { set_home_offset((AxisEnum)hmiValue.select, menuData.value / MINUNITMULT); }
   void setHomeOffsetX() { hmiValue.select = X_AXIS; setPFloatOnClick(-50, 50, UNITFDIGITS, applyHomeOffset); }
   void setHomeOffsetY() { hmiValue.select = Y_AXIS; setPFloatOnClick(-50, 50, UNITFDIGITS, applyHomeOffset); }
   void setHomeOffsetZ() { hmiValue.select = Z_AXIS; setPFloatOnClick( -2,  2, UNITFDIGITS, applyHomeOffset); }
```

**Comment by mriscoc** [comment]
NO_HOME_OFFSETS is a recent Marlin inclusion, you can check in Marlin documentation  as it stands to date: https://marlinfw.org/docs/gcode/M206.html and https://marlinfw.org/docs/gcode/M428.html  
  
![image](https://github.com/mriscoc/Ender3V2S1/assets/2745567/726ff06b-75f6-4c49-85f9-b262cbda478b)  

NO_WORKSPACE_OFFSETS was used before to disable M206:
https://github.com/MarlinFirmware/Marlin/blob/76030239283577ba1baca705e07182bb34345b64/Marlin/src/gcode/gcode.h#L193

HomeOffs.json is an old json and maybe will be removed soon, there is no reason to enable home offsets in the Professional firmware where you have a better control with XY MIN/MAX position, check here: https://github.com/mriscoc/Ender3V2S1/wiki/How-to-set-the-home-offsets
